File: Daily-contest/oct-14-25/3349.java
import java.util.List;

class Solution {
    public boolean hasIncreasingSubarrays(List<Integer> nums, int k) {
        int n = nums.size();
        for (int a = 0; a <= n - 2 * k; a++) {
            boolean firstIncreasing = true;
            for (int i = a; i < a + k - 1; i++) {
                if (nums.get(i) >= nums.get(i + 1)) {
                    firstIncreasing = false;
                    break;
                }
            }
            if (!firstIncreasing) continue;
            boolean secondIncreasing = true;
            for (int i = a + k; i < a + 2 * k - 1; i++) {
                if (nums.get(i) >= nums.get(i + 1)) {
                    secondIncreasing = false;
                    break;
                }
            }
            if (firstIncreasing && secondIncreasing) return true;
        }
        return false;
    }
}

File: Daily-contest/oct-14-25/3349.py
class Solution:
    def hasIncreasingSubarrays(self, nums: List[int], k: int) -> bool:
        n = len(nums)
        for a in range(n - 2 * k + 1):
            first_increasing = True
            for i in range(a, a + k - 1):
                if nums[i] >= nums[i + 1]:
                    first_increasing = False
                    break
            if not first_increasing:
                continue
            second_increasing = True
            for i in range(a + k, a + 2 * k - 1):
                if nums[i] >= nums[i + 1]:
                    second_increasing = False
                    break
            if first_increasing and second_increasing:
                return True
        return False
